### PR TITLE
Added protect/unprotect change feed events

### PR DESCRIPTION
**Pull request by ytqsl**
# Fixes or Implements [AB#15461](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15461)

[Task 15461](https://dev.azure.com/qslvic/Health%20Gateway/_workitems/edit/15461): Dependents - Wire in Service Bus to populate Message - Admin and Web

## Description

added protect/unprotect change feed events

## Testing

- [x] Unit Tests Updated
- [ ] Functional Tests Updated
- [ ] Not Required

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

**Review by BrianMaki — APPROVED** [review]

File: Apps/GatewayApi/src/appsettings.Development.json [hunk 1]
```diff
@@ -39,5 +39,10 @@
     "PhsaV2": {
         "TokenBaseUrl": "https://phsa-ccd-dev.azure-api.net/identity",
         "BaseUrl": "https://phsa-ccd-dev.azure-api.net/healthgateway"
+    },
+    "ChangeFeed": {
+        "Dependents": {
+            "Enabled": true
+        }
     }
```
Comment (MikeLyttle) [inline Apps/GatewayApi/src/appsettings.Development.json:47]: Shouldn't this be added to appsettings.json instead, and overridden here if needed?

File: Apps/Admin/Server/appsettings.Development.json [hunk 2]
```diff
@@ -39,5 +39,10 @@
     "PhsaV2Admin": {
         "TokenBaseUrl": "https://phsa-ccd-dev.azure-api.net/identity",
         "BaseUrl": "https://phsa-ccd-dev.azure-api.net/healthgatewayadmin"
+    },
+    "ChangeFeed": {
+        "Dependents": {
+            "Enabled": true
+        }
     }
```
Comment (MikeLyttle) [inline Apps/Admin/Server/appsettings.Development.json:47]: Shouldn't this be added to appsettings.json instead, and overridden here if needed?

File: Apps/Admin/Server/Services/DelegationService.cs [hunk 3]
```diff
@@ -202,22 +213,35 @@ public async Task<DelegationChange> ProtectDependentAsync(string dependentHdid,
...
 
             // Update dependent, allow delegation and resource delegate in database
-            await this.delegationDelegate.UpdateDelegationAsync(dependent, resourceDelegatesToDelete, dependentAudit).ConfigureAwait(true);
+            if (this.changeFeedEnabled)
+            {
+                await this.delegationDelegate.UpdateDelegationAsync(dependent, resourceDelegatesToDelete, dependentAudit, false);
+                var events = new MessageEnvelope[]
+                {
+                new MessageEnvelope(new DependentProtectionAddedEvent(dependentHdid), dependentHdid),
+                }.Concat(resourceDelegatesToDelete.Select(rd => new MessageEnvelope(new DependentRemovedEvent(rd.ProfileHdid, dependentHdid), rd.ProfileHdid)));
```
Comment (MikeLyttle) [inline Apps/Admin/Server/Services/DelegationService.cs:228]: Nit: indentation looks funky.

File: Apps/Admin/Server/Services/DelegationService.cs [hunk 4]
```diff
@@ -239,15 +263,29 @@ public async Task<DelegationChange> UnprotectDependentAsync(string dependentHdid
...
 
-            await this.delegationDelegate.UpdateDelegationAsync(dependent, Enumerable.Empty<ResourceDelegate>(), dependentAudit).ConfigureAwait(true);
+            if (this.changeFeedEnabled)
+            {
+                await this.delegationDelegate.UpdateDelegationAsync(dependent, Enumerable.Empty<ResourceDelegate>(), dependentAudit, false);
+
+                var events = new MessageEnvelope[]
+                {
+                        new MessageEnvelope(new DependentProtectionRemovedEvent(dependentHdid), dependentHdid),
+                };
```
Comment (MikeLyttle) [inline Apps/Admin/Server/Services/DelegationService.cs:273]: Nit: indentation looks funky.

File: Apps/GatewayApi/src/appsettings.Development.json (hunk 1 above)
Comment (ytqsl) [reply Apps/GatewayApi/src/appsettings.Development.json:47]: @sslaws and I discussed this, we'll add the flags to the common config in OCP to allow for explicitly setting this. However, it is not optional now, I'll make the code more tolerant when it is not configured.

File: Apps/Admin/Server/appsettings.Development.json (hunk 2 above)
Comment (ytqsl) [reply Apps/Admin/Server/appsettings.Development.json:47]: Same as above

File: Apps/GatewayApi/src/appsettings.Development.json (hunk 1 above)
Comment (MikeLyttle) [reply Apps/GatewayApi/src/appsettings.Development.json:47]: It looked like the code wouldn't fail it wasn't set, but would default to true.

File: Apps/Admin/Server/Services/DelegationService.cs (hunk 4 above)
Comment (ytqsl) [reply Apps/Admin/Server/Services/DelegationService.cs]: fixed

File: Apps/Admin/Server/Services/DelegationService.cs (hunk 3 above)
Comment (ytqsl) [reply Apps/Admin/Server/Services/DelegationService.cs]: fixed

File: Apps/GatewayApi/src/appsettings.Development.json (hunk 1 above)
Comment (ytqsl) [reply Apps/GatewayApi/src/appsettings.Development.json:47]: changed default to false

File: Apps/Admin/Server/Program.cs [hunk 5]
```diff
@@ -77,7 +81,7 @@ public static async Task Main(string[] args)
 
             Uri? baseUri = configuration.GetValue<Uri>("KeycloakAdmin:BaseUrl");
             services.AddRefitClient<IKeycloakAdminApi>()
-                .ConfigureHttpClient(c => c.BaseAddress = baseUri);
+                            .ConfigureHttpClient(c => c.BaseAddress = baseUri);
```
Comment (MikeLyttle) [inline Apps/Admin/Server/Program.cs:84]:
```suggestion
                .ConfigureHttpClient(c => c.BaseAddress = baseUri);
```

**Review by MikeLyttle — APPROVED** [review]

**Pull request merged**